src/ResourceManagement/EventHub/Ancestors.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using System;
+
+namespace Microsoft.Azure.Management.EventHub.Fluent
+{
+    class OneAncestor
+    {
+        private readonly String resourceGroupName;
+        private readonly String ancestor1Name;
+
+        public OneAncestor(String resourceGroupName, String ancestor1Name)
+        {
+            this.resourceGroupName = resourceGroupName ?? throw new ArgumentNullException("resourceGroupName");
+            this.ancestor1Name = ancestor1Name ?? throw new ArgumentNullException("ancestor1Name");
+        }
+
+        public OneAncestor(ResourceId resourceId)
+        {
+            if (resourceId == null)
+            {
+                throw new ArgumentNullException("resourceId");
+            }
+            if (resourceId.Id == null)
+            {
+                throw new ArgumentNullException("resourceId.iId");
+            }
+            if (resourceId.Parent == null)
+            {
+                throw new ArgumentNullException("resourceId.Parent");
+            }
+            this.resourceGroupName = resourceId.ResourceGroupName ?? throw new ArgumentNullException("resourceId.ResourceGroupName");
+            this.ancestor1Name = resourceId.Parent.Name ?? throw new ArgumentNullException("resourceId.Parent.Name");
+        }
+
+        public OneAncestor(String resourceId) : this(ResourceId.FromString(resourceId))
+        {
+        }
+
+        public String ResourceGroupName
+        {
+            get
+            {
+                return this.resourceGroupName;
+            }
+        }
+
+        public String Ancestor1Name
+        {
+            get
+            {
+                return this.ancestor1Name;
+            }
+        }
+    }
+
+    class TwoAncestor : OneAncestor
+    {
+        private String ancestor2Name;
+
+        public TwoAncestor(String resourceGroupName, String ancestor1Name, String ancestor2Name) 
+            : base(resourceGroupName, ancestor1Name)
+        {
+            this.ancestor2Name = ancestor2Name ?? throw new ArgumentNullException("ancestor2Name");
+        }
+
+        public TwoAncestor(ResourceId resourceId) : base(resourceId)
+        {
+            if (resourceId.Parent.Parent == null)
+            {
+                throw new ArgumentNullException("resourceId.Parent.Parent");
+            }
+            this.ancestor2Name = resourceId.Parent.Parent.Name ?? throw new ArgumentNullException("resourceId.Parent.Parent.Name");
+        }
+
+        public TwoAncestor(String resourceId) : this(ResourceId.FromString(resourceId))
+        {
+        }
+
+        public String Ancestor2Name
+        {
+            get
+            {
+                return this.ancestor2Name;
+            }
+        }
+    }
+
+    class ThreeAncestor : TwoAncestor
```

Comment:
>ThreeAncestor : TwoAncestor [](start = 10, length = 27)

I believe all these 3 inherited classes can be consolidated in one simple class with a type parameter which will identify if it is on, two or three ancestors, right?

Reply:
yes, as we discussed I created a work item to track this https://github.com/Azure/azure-libraries-for-java/issues/212.